BeNote/Home Tab/Main Screen/RightBarButtonManager.swift

```diff
@@ -25,83 +25,12 @@ extension HomeViewController{
                 action: #selector(onLogOutBarButtonTapped)
             )
             
-            navigationItem.rightBarButtonItems = [barIcon, barText]
-            
-        }else{
-            //MARK: not logged in...
-            let barIcon = UIBarButtonItem(
-                image: UIImage(systemName: "person.fill.questionmark"),
-                style: .plain,
-                target: self,
-                action: #selector(onSignInBarButtonTapped)
-            )
-            let barText = UIBarButtonItem(
-                title: "Sign in",
-                style: .plain,
-                target: self,
-                action: #selector(onSignInBarButtonTapped)
-            )
-            
-            navigationItem.rightBarButtonItems = [barIcon, barText]
-        }
-    }
-    
-    @objc func onSignInBarButtonTapped(){
-        let signInAlert = UIAlertController(
-            title: "Sign In / Register",
-            message: "Please sign in to continue.",
-            preferredStyle: .alert)
-        
-        //MARK: setting up email textField in the alert...
-        signInAlert.addTextField{ textField in
-            textField.placeholder = "Enter email"
-            textField.contentMode = .center
-            textField.keyboardType = .emailAddress
-        }
-        
-        //MARK: setting up password textField in the alert...
-        signInAlert.addTextField{ textField in
-            textField.placeholder = "Enter password"
-            textField.contentMode = .center
-            textField.isSecureTextEntry = true
-        }
-        
-        //MARK: Sign In Action...
-        let signInAction = UIAlertAction(title: "Sign In", style: .default, handler: {(_) in
-            if let email = signInAlert.textFields![0].text,
-               let password = signInAlert.textFields![1].text{
-                if (email.isEmpty || password.isEmpty) {
-                    self.showErrorAlert("Cannot submit empty fields")
-                }
-                //MARK: sign-in logic for Firebase...
-                self.signInToFirebase(email: email, password: password)
+            DispatchQueue.main.async {
```

Comment:
I was talking to the prof and idt we need the dispatch queue at all - basically it just makes a new thread to run whatever inside